modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -261,38 +309,151 @@ private PredicateExpressionVisitorImpl(
 		Map<String, String> lambdaVariableExpressionFieldNames,
 		long objectDefinitionId,
 		ObjectFieldBusinessTypeRegistry objectFieldBusinessTypeRegistry,
-		ObjectFieldLocalService objectFieldLocalService) {
+		ObjectFieldLocalService objectFieldLocalService,
+		ObjectRelatedModelsPredicateProviderRegistry
+			objectRelatedModelsPredicateProviderRegistry) {
 
-		_entityModel = entityModel;
+		_objectDefinitionsEntityModelMap.put(objectDefinitionId, entityModel);
```

Comment:
Ahh! I guess it is being this way because the objectDefinitionId passed by argument could be the related element.

My proposition would be one of these:

- Update the map with the new EntityModel if it does not exist for the following invocations
- Do something like:

```
private EntityModel _getObjectDefinitionEntityModel(
	long objectDefinitionId) {

	if (objectDefinitionId == _objectDefinition) {
		return _entityModel;
	}

	return _createEntityModel(objectDefinitionId);
}
```

Reply:
Oh God! 🤦 
My idea was to update the map with the new entityModel if it does not exist for the following invocations, but I forgot to update the map 🤦 

I should be more careful with these errors. 
Thanks!